pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -240,109 +241,220 @@ func substitueYamlVariables(baseYaml []byte, yamlSubstitutions map[string]string
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch deployment")
+	}
+	defer w.Stop()
+	success := make(chan *unstructured.Unstructured)
+	failure := make(chan error)
+
+	go func() {
+		defer close(success)
+		defer close(failure)
+
+		for {
+			val, ok := <-w.ResultChan()
+			if !ok {
+				failure <- errors.New("watch channel was closed")
+				return
+			}
+			if watchObject, ok := val.Object.(*unstructured.Unstructured); ok {
+				// TODO: Add more details on what to check to see if object deployment is complete
+				// Currently only checks to see if status.conditions[] contains a condition with type = conditionTypeValue
+				condition := getNamedCondition(watchObject, conditionTypeValue)
+				if condition != nil {
+					if condition["status"] == "Fail" {
+						failure <- fmt.Errorf("manifest deployment %s failed", applicationName)
+					} else if condition["status"] == "True" {
+						success <- watchObject
+					}
+				}
+			}
+		}
+	}()
+
+	select {
+	case val := <-success:
+		return val, nil
+	case err := <-failure:
+		return nil, err
+	case <-time.After(10 * time.Second):
+		return nil, errors.Errorf("timeout while waiting for %s manifest deployment completion", applicationName)
+	}
+}
+
 // Build image for devfile project
 func (a Adapter) Deploy(parameters common.DeployParameters) (err error) {
+
 	namespace := a.Client.Namespace
 	applicationName := a.ComponentName + "-deploy"
 	deploymentManifest := &unstructured.Unstructured{}
 
 	log.Info("\nDeploying manifest")
-	// TODO: Work out how to correctly handle spinners
-	s := log.Spinner("Deploying the manifest")
 
 	// Specify the substitution keys and values
 	yamlSubstitutions := map[string]string{
-		// TODO: this tag is not passed to delete, do we need to template
 		"CONTAINER_IMAGE": parameters.Tag,
-		"PROJECT_NAME":    applicationName,
+		"COMPONENT_NAME":  applicationName,
 		"PORT":            determinePort(parameters.EnvSpecificInfo),
 	}
 
-	// Substitute the values in the manifest file
-	deployYaml := substitueYamlVariables(parameters.ManifestSource, yamlSubstitutions)
-
 	// Build a yaml decoder with the unstructured Scheme
 	yamlDecoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
 
-	_, gvk, err := yamlDecoder.Decode([]byte(deployYaml), nil, deploymentManifest)
+	// This will override if manifest.yaml is present
+	writtenToManifest := false
+	manifestFile, err := os.Create(filepath.Join(a.Context, ".odo", "manifest.yaml"))
 	if err != nil {
-		return err
+		err = manifestFile.Close()
+		return errors.Wrap(err, "Unable to create the local manifest file")
 	}
 
-	klog.V(3).Infof("Deploy manifest:\n\n%s", deploymentManifest)
-	gvr := schema.GroupVersionResource{Group: gvk.Group, Version: gvk.Version, Resource: strings.ToLower(gvk.Kind + "s")}
-	klog.V(3).Infof("Manifest type: %s", gvr.String())
+	manifests := bytes.Split(parameters.ManifestSource, []byte("---"))
```

Comment:
what happens when the manifest only contains one resource? and we don't find the `---`, I suppose it returns the whole thing?

Reply:
Yes